src/wallet/wallet.cpp

```diff
@@ -2523,6 +2523,7 @@ bool CWallet::SelectCoins(const std::vector<COutput>& vAvailableCoins, const CAm
     {
         // For now, don't use BnB if preset inputs are selected. TODO: Enable this later
         bnb_used = false;
+        coin_selection_params.use_bnb = false;
```

Comment:
In commit "Actually disable BnB when there are preset inputs"

Sorry to repeat my comment from the other PR, but I do think it'd be clearer to add explicit non-reference `bool disable_bnb` parameters to SelectCoins methods, so the `CoinSelectionParams` struct can remain const and not be changing on the fly. The current bug seems like evidence to me that controlling bnb through the struct is not a great idea.

Reply:
I agree that it isn't ideal, but having the caller control whether BnB should be used is just a temporary shim to work with the current coin selection. I intend for it to go away when the replacement fallback strategy is implemented (which I have already started doing so). I don't think changing it now is worth the additional review effort required when the parameter will be removed later.

Reply:
Since this is just temporary code anyway, keep the parameter const and make a copy:

```c++
auto params = coin_selection_params;
params.use_bnb = false;
```

That way this diff doesn't have to change the function signature, you don't need to change the header file, and you can have more const. Considering all of the other work this method is doing, the overhead of the copy won't even be measurable.

Reply:
Actually, I reverted that change. It makes this untestable.